tools/harness-automation/gencsv.py

```diff
@@ -15,14 +38,101 @@ def get_key(item):
     return '%d-%s' % (no, ro)
 
 
-result = json.load(open(filename, 'r'))
-o = open('./result.csv', 'w')
-o.write('Case,Status,Started,Stopped,Reason\n')
-for k, v in sorted(result.items(), key=get_key):
-    o.write('%s,%s,%s,%s,%s\n' % (
-        k,
-        (v['passed'] and 'Pass') or 'Fail',
-        v['started'],
-        v['stopped'],
-        (v['error'] or '').replace('\n', ' '),
-    ))
+def get_case_id(scenario):
+    sg = scenario.split('_')
+    case_id = '.'.join(x for x in sg[1:])
+    return case_id
+
+
+def get_case_role(scenario):
+    sg = scenario.split('_')
+    return sg[0]
+
+
+def main():
+    parser = argparse.ArgumentParser(description='convert result.json to result.csv')
+    parser.add_argument(
+        '-t',
+        dest='test_date',
+        help='Date of testing',
+    )
+
+    parser.add_argument(
+        '-c',
+        dest='commit_id',
+        help='OpenThread commit id',
+    )
+
+    parser.add_argument(
+        '-v',
+        dest='harness_version',
+        help='Test Harness Version',
+    )
+
+    parser.add_argument(
+        '-d',
+        dest='dut',
+        help='DUT device',
+    )
+
+    parser.add_argument(
+        '-l',
+        dest='topo_file',
+        help='Topology configuration file name',
+    )
+
+    parser.add_argument(
+        '-f',
+        dest='result_json',
+        default='./result.json',
+        help='Result json file',
+    )
+
+    args = parser.parse_args()
+    result = json.load(open(args.result_json, 'r'))
+    with open('./result.csv', 'w') as o:
+        test_info = ''
+        empty_field = ''
+        if args.test_date:
+            o.write('Test Date,')
+            test_info += args.test_date + ','
+            empty_field += ','
+
+        if args.commit_id:
+            o.write('OT Commit ID,')
+            test_info += args.commit_id + ','
+            empty_field += ','
+
+        if args.harness_version:
+            o.write('Harness Version,')
+            test_info += args.harness_version + ','
+            empty_field += ','
+
+        if args.dut:
+            o.write('DUT,')
+            test_info += args.dut + ','
+            empty_field += ','
+
+        if args.topo_file:
+            o.write('Topology File,')
+            test_info += args.topo_file + ','
+            empty_field += ','
+
+        o.write('Test Case,DUT Role,Status,Logs,Started,Stopped\n')
+        if test_info is not None:
+            o.write('%s' % test_info)
+        for k, v in sorted(result.items(), key=get_key):
+            o.write('%s,%s,%s,%s,%s,%s\n' % (
+                get_case_id(k),
+                get_case_role(k),
+                (v['passed'] and 'Pass') or 'Fail',
+                (v['error'] or '').replace('\n', ' '),
+                v['started'],
+                v['stopped'],
```

Comment:
We don't need test start stop test time so we can drop these entries from the result.csv